### PR TITLE
Add styling when displaying enum values

### LINKED ISSUE #3799: No color styling when displaying enum values
**Describe the bug**
The read only view of enums does not contain colors

**To Reproduce**
1. Create a custom field with for a person of type _enum_ with choices containing colors, like:
```yml
            test:
              type: enum
              label: Project status
              choices:
                GREEN:
                  label: Green
                  color: '#c2ffb3'
                AMBER:
                  label: Amber
                  color: '#ffe396'
```
2. The person editor displays the colors
3. When viewing the person, the colors are not there

**Expected behavior**
`ReadonlyEnumField` should carry over colors like `EnumField` does

Other issue related to choice: #3804

### PR DESCRIPTION
**Pull request by anilsonmez-simsoft**
Colors of enum values are displayed on read-only views.

Closes #3799 

#### User changes
- Users can see the corresponding color of enum choices on read-only views.

#### Super User changes
- None

#### Admin changes
- None

#### System admin changes
- [ ] anet.yml or anet-dictionary.yml needs change
- [ ] db needs migration
- [ ] documentation has changed
- [ ] graphql schema has changed

### Checklist
  - [x] Described the user behavior in PR body
  - [x] Referenced/updated all related issues
  - [x] commits follow a `repo#issue: Title` title format and [these 7 rules](https://chris.beams.io/posts/git-commit/)
  - [x] commits have a [clean history](https://epage.github.io/dev/commits/), otherwise PR may be squash-merged
  - [ ] Added and/or updated unit tests
  - [ ] Added and/or updated e2e tests
  - [ ] Added and/or updated data migrations
  - [ ] Updated documentation
  - [ ] Resolved all build errors and warnings
  - [ ] Opened debt issues for anything not resolved here

### CODE REVIEW

File: client/src/components/CustomFields.js [hunk 1]
```diff
@@ -286,9 +286,42 @@ const EnumField = fieldProps => {
 
 const enumHumanValue = (choices, fieldVal) => {
```
Comment (midmarch) [inline client/src/components/CustomFields.js:287]: Not part of the change, maybe good to directly update:
```js
const enumHumanValue = () => {}
function enumHumanValue() {} // Shorter and less operators
```
Other motivation: some say that it as adventages in debugging as the name of the function stays preserved.

File: client/src/components/CustomFields.js [hunk 2]
```diff
@@ -286,9 +286,42 @@ const EnumField = fieldProps => {
...
+                fontWeight: "inherit",
+                lineHeight: "inherit",
+                color: "black",
+                backgroundColor: choices[k]?.color
+              }}
+            >
+              {choices[k]?.label}
+            </Badge>
+          ))
+          .reduce((acc, x) => (acc === null ? [x] : [...acc, ", ", x]), null)}
```
Comment (asafkaganbezgin) [inline client/src/components/CustomFields.js:306]: I think using Array.reduce() for adding a comma seperator is a bit expensive. The addition of comma can take place during the first iteration (Array.map) like the following:
```
{fieldVal.map((k, index) => (
  <>
    <Badge
      key={k}
      style={{
        fontSize: "inherit",
        fontWeight: "inherit",
        lineHeight: "inherit",
        color: "black",
        backgroundColor: choices[k]?.color
      }}
    >
      {choices[k]?.label}
    </Badge>
    {index < fieldVal.length - 1 && " , "}
  </>
))}
```

Comment (midmarch) [inline client/src/components/CustomFields.js:306]: Maybe good comment this piece of magic it to sperate the elements with `", "`

**Review by asafkaganbezgin — CHANGES_REQUESTED** [review]

**Review by midmarch — APPROVED** [review]

**Review by asafkaganbezgin — APPROVED** [review]
